test/test.make

```diff
@@ -0,0 +1,53 @@
+THIS_FILE := $(lastword $(MAKEFILE_LIST))
+BINARIES= builds/${BUILD_TARGET}/luajit builds/${BUILD_TARGET}/libluajit.so
+BUILDS= normal gc64 dualnum nojit
+
+normal : XCFLAGS= 
+normal : export BUILD_TARGET=normal
+gc64 : export XCFLAGS=-DLUAJIT_ENABLE_GC64
+gc64 : export BUILD_TARGET=gc64
+dualnum : export XCFLAGS=-DLUAJIT_NUMMODE=2
+dualnum : export BUILD_TARGET=dualnum
+nojit : export XCFLAGS=-DLUAJIT_DISABLE_JIT
+nojit : export BUILD_TARGET=nojit
+
+build_luajit:
+	@echo "==== Building LuaJIT target = ${BUILD_TARGET} ===="
+	$(MAKE) -C ../src -j XCFLAGS="-DLUAJIT_ENABLE_LUA52COMPAT ${XCFLAGS}"
+	mkdir -p builds/${BUILD_TARGET}/
+	cp ../src/luajit builds/${BUILD_TARGET}/
+	cp ../src/libluajit.so builds/${BUILD_TARGET}/
+	$(MAKE) -C ../src clean
+	@echo "==== Successfully built LuaJIT ===="
+
+$(BINARIES) : build_luajit|
+
+build_binaries : $(BINARIES)
+
+#Recursive call so we can expand ${BUILD_TARGET} in top level varaibles
+normal gc64 dualnum nojit: 
+ifeq (,$(BUILD_TARGET))
+	@$(MAKE) -f $(THIS_FILE) BUILD_TARGET=${BUILD_TARGET} build_binaries

```

Comment:
I might be an idiot, but I can't see why we need this. You can expand variables in other ways IIRC.

Reply:
I used [https://www.gnu.org/software/make/manual/make.html#Target_002dspecific](Target-specific Variable) to avoid repeating tons of stuff but I couldn't find a way to delay expanding it in the $(BINARIES) : build_luajit' target.

Reply:
Don't you want something like (untested!):

```make
ifeq($TARGET, "normal")
  XCFLAGS= 
  BUILD_TARGET=normal
fi
```

and so on?

Reply:
Basically, I'm 99% certain there's a nicer way than the recursive make. I had to stare at this for ages to work out what was supposed to be going on, and I'm not a complete dummy when it comes to make!

Reply:
That doesn't handle when you want to build all of them and ifeq is only evaluated at the top level

Reply:
Then maybe you need to use a `for` loop somewhere. Basically, unless I'm totally convinced that the only possible route is recursive make (and that's going to take a *lot* of convincing ;) ), I'm going to ask for another route.

Reply:
I managed to get all build target working but i wanted to use the commented out lines to build it but they cause it to always rebuild even when its not out of date